src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -1180,7 +1180,7 @@ private void SaveCore(ModelSaveContext ctx)
             ctx.Writer.Write(sizeof(float));
             ctx.Writer.Write(BinSize);
             ctx.Writer.Write(Min);
-            ctx.Writer.WriteSingleArray(BinProbs);
+            ctx.Writer.WriteSingleArray(BinProbs as float[]);
```

Comment:
> as float[] [](start = 48, length = 11)

Let's not please... let's do a backing private field of `float[]` that we actually operate over. This will have a couple of advantages, most notably not having this bad code smell, but also incidentally making access to this structure much faster since it won't have to work through the virtual method table on every single usage.

Reply:
Just to make completely clear.
We will have:
```
public IReadOnlyList<float> BinProbs => _binProbs;
private readonly float[] _binProbs;
```
And for loading and saving we will use `_binProbs`
Does it sound good?

Reply:
And other operators too, yes.